lib/iris/experimental/regrid_conservative.py

```diff
@@ -274,6 +274,10 @@ def _valid_units(coord):
         # Paste regridded slice back into parent array
         fullcube_data[slice_indices_tuple] = data
 
+    # Remove the data mask if completely unused.
+    if not np.ma.is_masked(fullcube_data):
```

Comment:
This array is constructed as a masked array at [line 223](https://github.com/SciTools/iris/pull/2856/commits/f63d9501f0f23ddb348fa39bff35c37753f8b59b#diff-37a99e4d6154ed1e61924c68435ecfb5R223), so if we don't convert it to an ndarray it will output a masked array even if the input is an ndarray.